ichnaea/locate/provider.py

```diff
@@ -609,24 +645,37 @@ def locate(self, data):
                     verify=False,
                 )
             self.stat_count('fallback.lookup_status.%s' % response.status_code)
-            if response.status_code == 404:
+            if response.status_code != 404:
                 # don't log exceptions for normal not found responses
-                response = None
-            else:
```

Comment:
@jaredkerim if you'd add this `response = None` back, that should result in the "cache not found" test to break, right?

What happens now is that we store the large nested JSON dict response from the fallback service into the cache. And than later on we try to parse that and get `response['location']['lat']` and friends from it, which fails and triggers a KeyError. That's not really all that nice and we trigger a raven error message each time. If you add a `self.check_raven(total=0)` to the test, that should fail right now and log this KeyError.

I'd rather make this explicit and create a special token value for not found and check for that one.